xonsh/procs/specs.py

```diff
@@ -737,7 +737,9 @@ def _update_last_spec(last):
         return
     callable_alias = callable(last.alias)
     if callable_alias:
-        pass
+        if last.cls is ProcProxy and captured == "hiddenobject":
+            # a ProcProxy run using ![] should not be captured
```

Comment:
I think it can short circuit for other non-capturing types as well. like $[]

Reply:
`$[]` (`captured=False`) is already short-circuited at the start of the function:
https://github.com/xonsh/xonsh/blob/09f3a4ea143907f889d5b40b4143caf7363000aa/xonsh/procs/specs.py#L736-L737